node/src/service.rs

```diff
@@ -242,16 +243,27 @@ pub fn new_full(config: Configuration) -> Result<TaskManager, ServiceError> {
                 select_chain,
                 block_import,
                 proposer_factory,
-                create_inherent_data_providers: move |_, ()| async move {
-                    let timestamp = sp_timestamp::InherentDataProvider::from_system_time();
-
-                    let slot =
-						sp_consensus_aura::inherents::InherentDataProvider::from_timestamp_and_slot_duration(
-							*timestamp,
-							slot_duration,
-						);
-
-                    Ok((slot, timestamp))
+                create_inherent_data_providers: move |parent_hash, ()| {
+                    let parent_block = client_for_cidp
+                        .clone()
+                        .block(parent_hash)
+                        .expect("a")
```

Comment:
It took a while, but I found where this is used. 
https://github.com/paritytech/polkadot-sdk/blob/7aace06b3d653529ab992c8ff96571d54bf00a36/substrate/client/consensus/slots/src/slots.rs#L154 
It logs the error and tries again next slot, so it might not like a panic

Reply:
Definitely we will not be able to author a block if the database does not know about the previous block. What do you suggest to improve this?

Reply:
I was trying earlier to make this return an error when needed, instead of panicking. I wanted to do someting like:
```rust
let parent_block = client_for_cidp
    .clone()
    .block(parent_hash)?
    .ok_or(sp_blockchain::Error::UnknownBlock(parent_hash))?
    .block;
```
There was some issues with the closure being async, and I'm not familiar with that, so please try to make this work without spending too much time on it, otherwise it's fine to leave it like this.